src/views/components/response_components/select_ta.vue

```diff
@@ -0,0 +1,152 @@
+                            return response.json();
+                        })
+                        .then(data => {
+                            this.ta_package = this.process_tas(data);
+                            this.all_terms = [201709, 201801, 201701];
+                            if (!this.course) {
+                                this.all_courses = Object.keys(this.ta_package);
+                            }
+                        })
+                        .catch(err => {
+                            this.$emit("error", err.toString());
+                        });
+                    break;
+                case "ta":
+                    this.target_ta = user_id;
+                    this.select_ta(
+                        user_id,
+                        this.user_name,
+                        term,
+                        course,
+                        user_id
+                    );
+                    break;
+                default:
+                    console.warn(
+                        "`user_type` of ",
+                        user_type,
+                        "is unrecognized."
+                    );
+            }
+        },
+        select_ta: function(ta_id, ta_name, term, course, user_id) {
+            this.$router.push({
+                name: "response_page",
+                query: {
+                    target_ta: ta_id,
+                    term: term,
+                    course: course,
+                    user_id: user_id,
+                    selected_ta: ta_name
+                }
+            });
+        },
+        // group ta by courses
+        // Input: {DATA: [{course_code: "CSC100", name: "Daniel", user_id: "ta1"},
+        //               {course_code: "CSC100", name: "Darren", user_id: "ta2"},
+        //               {course_code: "CSC101", name: "Daniel", user_id: "ta1"},
+        //               {course_code: "CSC101", name: "Darren", user_id: "ta2"}
+        //  ]}
+        // Output : { CSC100: [{name: "Daniel", id: "ta1"},{name: "Dareen", id: "ta2"}]
+        //            CSC101: [{name: "Daniel", id: "ta1"},{name: "Dareen", id: "ta2"}]
+        //}
+        process_tas: function(data) {
+            let ta_package = {};
+            for (let row of data.DATA) {
+                ta_package[row.course_code] = ta_package[row.course_code] || [];
+                ta_package[row.course_code].push({
+                    name: row.name,
+                    id: row.user_id
+                });
+            }
+            return ta_package;
+        },
+        // return false if object is null or empty
+        isEmpty: function(object) {
+            return !object || Object.keys(object).length === 0;
+        }
+    },
+    computed: {
+        //returns the updated component of term and course combined
+        // added date to be unique every time
+        term_and_course() {
+            this.term;
+            this.course;
```

Comment:
I still don't believe you. Are you claiming that `this.term` is a dynamically computed property? Those are very uncommon in JS...

Reply:
`this.term` isn't a dynamically computed property, `term_and_course` is. Vue creates this dependency relationship such that it knows `term_and_course` depends on `this.term` and `this.course`, and thus any changes to it's dependencies will trigger the function. As the Vue documentation states "A computed property will only re-evaluate when some of its dependencies have changed.".

Reply:
Then removing these lines shouldn't make a difference. Can you confirm this?

Reply:
The entire function? Also dependencies must to be reactive, which means that Date.now() by itself will never trigger an update. I tried with different combinations within this function, this is the one works best if you didn't want comma separated return statements.

Reply:
This cannot be right. Figure out what is actually going on and what is actually needed. As it is, you have two NOOP  (https://en.wikipedia.org/wiki/NOP) instructions followed by a return statement. The NOOPs shouldn't be there.

Reply:
After a little bit of digging into Vue's source code and community, here's what I got out of it: 
1. All `data` properties are defined as getters and setters in Vue, which are also considered `reactive' properties of the particular Vue instance.
2. On declaration of a computed property, Vue associates all 'reactive' getters in the computed property into the dependency list of this computed property (supported by the `Dep` dependency tracker in Vue source code). Which allows the computed property to watch the list for change.
3. All properties requesting the computed property will also add the computed property to it's dependency list. 
4. Despite the variables are noop, they are just used associated with the dependency list on initialization of the computed property.
5. Vuejs creator commented on another way to do this without having to use the computed property *cries*. but I find it as an easy way to collectively watch a set of variables we use for filtering.

Reply:
Okay. Add a comment saying that these are accessed via getters and therefore have sideeffects in Vue and that this is on purpose with a link to the discussion you read.

Reply:
If this makes no sense, I will switch to only using the `watch` property.